package/MDAnalysis/core/_get_readers.py

```diff
@@ -176,6 +177,14 @@ def get_writer_for(filename, format=None, multiframe=None):
                 None)
         else:
             format = util.check_compressed_format(root, ext)
+        
+    if format == '':
+        # Improves detail of error message when ext is not assigned in
+        # output file. Otherwise TypeError is raised when except the
+        # return statement
+        raise FileIOError((
```

Comment:
This should  also be `ValueError` as the format could not be guessed.

If the format was set by the user as empty string then that should be an error (also `ValueError`) as there is no way to determine a format (it enables auto-detection (`None`) and does not provide a valid format).

Reply:
This happens under the circumstances described in the PR message:

```python
with mda.Writer('output_traj', selection=selection.n_atoms) as W:
    for ts in u.trajectory:
        W.write(selection)
  File "/home/joao/GitHub/fork_mdanalysis/package/MDAnalysis/coordinates/core.py", line 134, in writer
    multiframe=kwargs.pop('multiframe', None))
  File "/home/joao/GitHub/fork_mdanalysis/package/MDAnalysis/core/_get_readers.py", line 199, in get_writer_for
    raise_from(TypeError(errmsg.format(format)), None)
  File "<string>", line 3, in raise_from
TypeError: No trajectory or frame writer for format ''
```